### DISCUSSION TITLE
x-for outputs duplicate values

### DISCUSSION THREAD

**Discussion by Solotov** [alpinejs/alpine#2036] [7. Bugs (archived)]
## X-FOR  in alpinejs version  3.2.4 

**all examples on the page produce duplicate results :**
[https://alpinejs.dev/directives/for](https://alpinejs.dev/directives/for)

for example the first example on the previous page produces :

- Red
- Orange
- Yellow
- Red
- Orange
- Yellow

I have switched to working with version **2.8.2** and everything seems to be working fine.

**Comment by SimoTod** [comment]
They example seems to work for me. 
If that had been the case it would have been flagged ages ago, there are a lot of people using Alpine daily. Are you starting Alpine twice?

**Reply by Solotov** [reply]
I only call "**Alpine.start()**" once in my code,

when I comment that line of code, alpine no longer works.

I tried it in jsfiddle and it gives the same error:
[https://jsfiddle.net/solotov/2v0zpk4u/5/](https://jsfiddle.net/solotov/2v0zpk4u/5/)

**Reply by SimoTod** [reply]
mmmh, that example doesn't follow the instructions.
1) if you use the cdn version, you shouldn't call Alpine.start(). Alpine.start is just when you bundle Alpine into your compiled script using npm, etc etc
2) Alpine script must be deferred, don't use the resource tab of jsfiddle because that won't defer the script and it will add Alpine at the top of the page, causing this sort of errors. (Technically it will work just fine without the deferred attribute if it's at the bottom of the page).

See https://jsfiddle.net/qebwcahd/2/

The reason is that Alpine v2 was waiting for the page ready event while v3 doesn't and it starts as soon as the script tag is parsed. The defer attribute makes sure that the script is parsed once the whole page has been downloaded but, without it, Alpine kicks in when the page is still incomplete and the mutation observer will cause a lot of troubles while all the single HTML tags are parsed causing duplications and errors.

N.b your last example didn't have an x-data attribute

To recap
without defer attribute only works at the bottom of the page (still not advised)
with defer attribute, okay everywhere

**Reply by Solotov** [reply]
Yes, you are right... in fact I am new in this framework,
Thanks now it works! !

**Reply by SimoTod** [reply]
No worries 👍
